lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2693,83 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Signature for [PlatformDispatcher.onViewFocusChange].
+typedef ViewFocusChangeCallback = void Function(ViewFocusEvent viewFocusEvent);
+
+/// An event for the engine to communicate view focus changes back to the app.
+///
+/// This value will be tipically passed to the [PlatformDispatcher.onViewFocusChange]
+/// callback.
+final class ViewFocusEvent {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusEvent({
+    required this.viewId,
+    required this.focusState,
+    required this.focusDirection,
+  });
+
+  /// The id of the [FlutterView] that experienced a focus change.
+  final int viewId;
+
+  /// The current [ViewFocusState] of the [FlutterView] with the given [viewId].
+  final ViewFocusState focusState;
+
+  /// The current [ViewFocusDirection] of the [FlutterView] with the given [viewId].
+  final ViewFocusDirection focusDirection;
+
+  @override
+  String toString() {
+    return 'ViewFocusEvent(viewId: $viewId, focusState: $focusState, focusDirection: $focusDirection)';
+  }
+}
+
+/// Represents the focus state of a given [FlutterView].
+///
+/// This enum will be associated with a [FlutterView] by instantiating a [ViewFocusEvent].
+///
+/// If focus was lost, use [ViewFocusState.unfocused].
+/// If focus was gained, use [ViewFocusState.focused].
+///
+/// Valid transitions, within a view, are:
+///
+/// - [ViewFocusState.focused] to [ViewFocusState.unfocused].
+/// - [ViewFocusState.unfocused] to [ViewFocusState.focused].
+///
+/// See also:
+///
+///   * [ViewFocusDirection], that specifies the focus direction.
+///   * [ViewFocusEvent], that conveys information about a [FlutterView] focus change.
+enum ViewFocusState {
+  /// Specifies that the given flutter view gained focus.
+  unfocused,
+
+  /// Specifies that the given flutter view lost focus.
+  focused,
+}
+
+/// Represents the focus direction  of a given [FlutterView].
```

Comment:
The doc implies that a FlutterView _has_ a focus direction. However, direction only applies to a transition/event. Once the transition is executed the direction is forgotten. Consider something like this:

```dart
/// The direction in which a view gained or lost focus.
```